networkx/algorithms/isomorphism/tree_isomorphism.py

```diff
@@ -1,29 +1,1021 @@
+    for (parent, child) in nx.bfs_edges(T, root):
+        parenthood[child] = parent
+
+    return parenthood
+
+
+def rooted_tree_isomorphism_n(T1, root_T1, T2, root_T2):
+    """Returns an isomorphism between two rooted trees, if it exists.
+
+    Parameters
+    ----------
+    T1 : NetworkX Graph
+        A rooted tree.
+
+    root_T1 : node
+        The root of the rooted tree T1.
+
+    T2 : NetworkX Graph
+        A rooted tree.
+
+    root_t2 : node
+        The root of the rooted tree T2.
+
+    Returns
+    -------
+    are_iso : bool
+        Returns True if the given rooted trees are isomorphic, False otherwise.
+
+    isomorphism : dict of node of T1: node of T2
+        The isomorphism mapping f such that (u, v) is in E_T1 if and only if
+        (f(u), f(v)) is in E_T2. This dictionary is empty when there is no
+        isomorphism between the trees.
+
+    Note
+    ----
+    This algorithm runs in O(n) time and space.
+
+    """
+    # If both trees have different amount of vertices, return false.
+    if T1.order() != T2.order():
+        return False, {}
+
+    # If both trees are empty, return true.
+    if T1.order() == 0:
+        return True, {}
+
+    # If trees have more than one node, verify that they're indeed trees.
+    assert nx.is_tree(T1)
+    assert nx.is_tree(T2)
+
+    # Get the height's of the trees.
+    height_T1 = get_height(T1, root_T1)
+    height_T2 = get_height(T2, root_T2)
+
+    # If the trees differ in height, return false.
+    if height_T1 != height_T2:
+        return False, {}
+    elif height_T1 == 0:
+        # If both trees have height 0, then there are only conformed by the
+        # roots, they're isomorph.
+        return True, {root_T1: root_T2}
+    else:
+        # If they have the same amount of levels, check that all levels coincide
+        # in structure.
+        levels_T1 = get_levels(T1, root_T1, height_T1)
+        levels_T2 = get_levels(T2, root_T2, height_T2)
+
+        # For every level i, check that both trees have the same amount of
+        # vertices on the i-th level. If they differ for some level, return
+        # false.
+        for i in range(height_T1):
+            if len(levels_T1[i]) != len(levels_T2[i]):
+                return False, {}
+
+        # Set the initial values for T1 and T2.
+        values_T1 = get_initial_values(T1, root_T1)
+        values_T2 = get_initial_values(T2, root_T2)
+
+        # Define the parenthood map for T1 and T2.
+        parenthood_T1 = get_parenthood(T1, root_T1)
+        parenthood_T2 = get_parenthood(T2, root_T2)
+
+        # Check that all levels have the same structure.
+        return levels_verification(
+            root_T1,
+            root_T2,
+            values_T1,
+            values_T2,
+            levels_T1,
+            levels_T2,
+            parenthood_T1,
+            parenthood_T2,
+            height_T1,
+        )
+
+
+def get_centers_of_tree(T):
```

Comment:
This is a nice function that's faster than `nx.center` for trees; it might be useful to move it next to `nx.center` and make it available to the API (as, idk, `nx.tree_center` ?) 🙂